specs/trip_dispatch_spec.rb

```diff
@@ -1,114 +1,200 @@
+
+
+            expect(first_passenger.name).must_equal "User1"
+            expect(first_passenger.id).must_equal 1
+            expect(last_passenger.name).must_equal "Driver8"
+            expect(last_passenger.id).must_equal 8
+          end
+
+          it "accurately loads trip info and associates trips with passengers" do
+            trip = @dispatcher.trips.first
+            passenger = trip.passenger
+
+            expect(passenger).must_be_instance_of RideShare::User
+            expect(passenger.trips).must_include trip
+          end
+        end
+
+        describe "#request_trip(user_id)" do
+          before do
+            @dispatcher = RideShare::TripDispatcher.new(USER_TEST_FILE,
+              TRIP_TEST_FILE, DRIVER_TEST_FILE)
+            end
+            # let (:new_dispatcher) {
+            #   dispatcher = RideShare::TripDispatcher.new(USER_TEST_FILE,
+            #                                               TRIP_TEST_FILE, DRIVER_TEST_FILE)
+            #   dispatcher.request_trip(5)
+            #
+            # }
+            it "will return an instance of trip" do
+              expect(@dispatcher.request_trip(4)).must_be_instance_of RideShare::Trip
+            end
+
+            it "will raise error if user does not exist" do
+              expect {@dispatcher.request_trip(40000)}.must_raise ArgumentError
+            end
+
+            it "returns a driver who becomes UNAVAILABLE" do #when we request a driver it should become unavailable
+              expect(@dispatcher.request_trip(1).driver.status).must_equal :UNAVAILABLE
+            end
+
+            it "raises argument error if all drivers are unavailable" do
+              @dispatcher.request_trip(1)
+              @dispatcher.request_trip(5)
+              expect{@dispatcher.request_trip(8)}.must_raise ArgumentError
+            end
+
+            it "increases the trips in the driver_driven_trips array" do
+
+              driver = @dispatcher.drivers.find { |driver| driver.id == 8 }
+              before_new_trip = driver.driven_trips.length
+
+              @dispatcher.request_trip(7)
+
+              # driver = @dispatcher.drivers.find { |driver| driver.id == 8 }
+              # before_new_trip = driver.driven_trips.length
+              @dispatcher.request_trip(4)
+              after_new_trip = driver.driven_trips.length
+              expect(before_new_trip < after_new_trip).must_equal true
+            end
+
+            it "increases the number trips in the users list of trips array" do
+              @dispatcher.request_trip(5)
+              passenger = @dispatcher.passengers.find { |passenger| passenger.id ==  1}
+              before_new_trip = passenger.trips.length
+              @dispatcher.request_trip(1)
+              after_new_trip = passenger.trips.length
+              expect(before_new_trip < after_new_trip).must_equal true
+            end
+
+            it "makes sure driver is not driving themself " do
```

Comment:
I would add one test where there is one driver available and that driver requests a trip.